src/compiler/checker.ts

```diff
@@ -9636,14 +9636,24 @@ namespace ts {
                     outer: for (const t of targetSignatures) {
                         // Only elaborate errors from the first failure
                         let shouldElaborateErrors = reportErrors;
+
+                        // Only consider source signatures which are maximal with regard to the target signature
+                        let maximumEffectiveCallLength = 0;

```

Comment:
Calculating maximumEffectiveCallLength every time so we don't have to allocate an array?